src/components/tl/ucp/tl_ucp_context.c

```diff
@@ -106,13 +109,46 @@ UCC_CLASS_INIT_FUNC(ucc_tl_ucp_context_t,
                  "failed to initialize tl_ucp_req mpool");
         goto err_thread_mode;
     }
-    if (UCC_OK != ucc_context_progress_register(
-                      params->context,
+    ucc_status = ucc_context_progress_register(params->context,
                       (ucc_context_progress_fn_t)ucp_worker_progress,
-                      self->ucp_worker)) {
+                      self->ucp_worker);
+    if (UCC_OK != ucc_status) {
         tl_error(self->super.super.lib, "failed to register progress function");
         goto err_thread_mode;
     }
+    if ((params->params.mask & UCC_CONTEXT_PARAM_FIELD_OOB) &&
+        (params->params.mask & UCC_CONTEXT_PARAM_FIELD_EP) &&
+        (params->params.mask & UCC_CONTEXT_PARAM_FIELD_FLAGS) &&
+        (params->params.flags & UCC_CONTEXT_FLAG_TEAM_EP_MAP)) {
+        if (params->params.ep >= params->params.oob.participants) {
+            tl_error(self->super.super.lib,
+                     "incorrect ctx ep %llu: out of oob.participants range %u",
+                     (long long unsigned)params->params.ep,
+                     params->params.oob.participants);
+            ucc_status = UCC_ERR_INVALID_PARAM;
+            goto err_thread_mode;
```

Comment:
Need one more label to unregister progress function or move ucc_context_progress_register to the end of function